### DISCUSSION TITLE
JSX factory polymorphic "as" prop

### DISCUSSION THREAD

**Discussion by jacoblapworth** [chakra-ui/panda#1000] [General]
Similar to libraries like Stitches, it can be useful to change the html tag of a component in it's implementation.

For example you might have a heading component where you want to separate the semantic heading level, from the visual appearance: 

```jsx
<Heading as="h1" size="lg">Title</Heading>
```

It seems like this actually works under the hood with the `styled` asx factory, but it's not exposed in the types
```
Property 'as' does not exist on type 'IntrinsicAttributes & Omit<WithHTMLProps<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>>, "base" | ... 797 more ... | "css">
```

Is this a bug / yet to be implemented?

**Comment by anubra266** [comment]
@jacoblapworth We're not implementing `as` cause we don't intend for panda to have to deal with the complexities involved in this. So you can take Panda as purely a styling engine, but not something to handle component complexities.
